app/src/main/java/com/hedvig/app/feature/addressautocompletion/data/GetDanishAddressAutoCompletionUseCase.kt

```diff
@@ -0,0 +1,44 @@
+package com.hedvig.app.feature.addressautocompletion.data
+
+import arrow.core.Either
+import com.apollographql.apollo.ApolloClient
+import com.hedvig.android.owldroid.graphql.AddressAutocompleteQuery
+import com.hedvig.android.owldroid.type.AddressAutocompleteType
+import com.hedvig.app.feature.addressautocompletion.model.DanishAddress
+import com.hedvig.app.feature.addressautocompletion.model.DanishAddressInput
+import com.hedvig.app.util.apollo.QueryResult
+import com.hedvig.app.util.apollo.safeQuery
+
+class GetDanishAddressAutoCompletionUseCase(
+    private val apolloClient: ApolloClient,
+) {
+    suspend operator fun invoke(
+        input: DanishAddressInput,
+    ): Either<QueryResult.Error, AddressAutoCompleteResults> {
+        val addressAutocompleteQuery = AddressAutocompleteQuery(
+            input.queryString,
+            input.toAddressAutocompleteType()
+        )
+        return apolloClient
+            .query(addressAutocompleteQuery)
+            .safeQuery()
+            .toEither()
+            .map { queryData ->
+                AddressAutoCompleteResults(queryData.autoCompleteAddress.map(DanishAddress::fromDto))
+            }
+    }
+}
+
+data class AddressAutoCompleteResults(
+    val resultList: List<DanishAddress>,
+)
+
+private fun DanishAddressInput.toAddressAutocompleteType(): AddressAutocompleteType {
```

Comment:
The API needs a different type depending on which stage of the querying we are currently in. This method maps the current state of the Address we're looking for, to the type that needs to be queried